slmigrate/mongohandler.py

```diff
@@ -25,39 +26,82 @@ def stop_mongo(proc):
 def capture_migration(service, action, config):
     if action != constants.capture_arg:
         return
-    # mongo_migration_dir = os.path.join(service.migration_dir, "mongo-dump")
-    # mongo_migration_dir = os.path.join(service.migration_dir, "mongo-dump")
     cmd_to_run = constants.mongo_dump + " --port " + str(config[service.name]['Mongo.Port']) + " --db " + config[service.name]['Mongo.Database'] + " --username " + config[service.name]['Mongo.User'] + " --password " + config[service.name]['Mongo.Password'] + " --out " + constants.mongo_migration_dir + " --gzip"
     subprocess.run(cmd_to_run)
 
 
 def restore_migration(service, action, config):
     if action != constants.restore_arg:
         return
-    # mongo_migration_dir = os.path.join(service.migration_dir, "mongo-dump")
     mongo_dump_file = os.path.join(constants.mongo_migration_dir, config[service.name]['Mongo.Database'])
     cmd_to_run = constants.mongo_restore + " --port " + str(config[service.name]['Mongo.Port']) + " --db " + config[service.name]['Mongo.Database'] + " --username " + config[service.name]['Mongo.User'] + " --password " + config[service.name]['Mongo.Password'] + " --gzip " + mongo_dump_file
     subprocess.run(cmd_to_run)
 
 
+def migrate_document(destination_collection, document):
+    try:
+        print("Migrating " + str(document['_id']))
+        destination_collection.insert_one(document)
+    except pyerr.DuplicateKeyError:
```

Comment:
Is there a case where you would expect we'd see a duplicate key, or is this more defensive?

If there's a case where we'd expect to see a duplicate key, that's data that potentially didn't get migrated, right?  In that case we might want these messages to be more prominent than the other logging.  And it might be good to include the collection name in the message so that from the log we know exactly where to go look for data that might need to be manually merged.

Reply:
This is all about being defensive in case the tool gets run more than once, or if the user found themselves in this scenario more than once. It will exit otherwise and we might still need to do migration/merging downstream of this call.